public/data.js

```diff
@@ -27,19 +27,37 @@ class DataController {
             // Update data every time it changes in firestore
             DataController.updateData(res, activeProjects);
             onChange(activeProjects);
+        }, error => {
+            if (error.code == "permission-denied") {
+                firebase.auth().onAuthStateChanged(user => { 
+                    if (user) {
+                        name = user.displayName.split(" ").slice(0,-1)
```

Comment:
As much as this is nice to have let minimize the logic as this eventually takes a toll on the dashboard performance.

Reply:
Thanks, updated.